app/mm-redux/utils/group_utils.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import {General} from '../constants';
+import {Group} from '@mm-redux/types/groups';
+import {getSuggestionsSplitByMultiple} from './user_utils';
+
+export function filterGroupsMatchingTerm(groups: Array<Group>, term: string): Array<Group> {
```

Comment:
Have we benchmarked this utility function? it is doing a lot of stuff

Reply:
Here are my results. I wrote this code to benchmark in the `group_utils.test.js` file
```Javascript
describe('benchmark group utils', () => {
    const entries = 100000;
    const runTimes = 100;
    let groups = [];
    let totalTime = 0;
    for (let i = 0; i < entries; i++) {
        const group = {
            name: `software-engineers-${i}`,
            display_name: `software engineers ${i}`,
            allow_reference: true,
        };  
        groups.push(group);
    }

    for (let i = 0; i < runTimes; i++) {
        const then = new Date;
        filterGroupsMatchingTerm(groups, '@software')
        const now = new Date;
        totalTime += (now - then);
    }

    console.log(`Average run time for ${entries} groups ran ${runTimes} times is: ${totalTime / runTimes} ms`);
});
```

```
Average run time for 10,000 groups ran 100 times is: 39.03 ms

Average run time for 100,000 groups ran 100 times is: 434.14 ms

Average run time for 250,000 groups ran 100 times is: 1084.11 ms
```

Hope that's good!

Reply:
for  10K is fine.. 100k is OK... 250k is definitely bad (for a case like this it would probably be best to paginate it or apply some sort of limit)

I don't know if hitting 250K groups is something that we are expecting, if not just disregard this comment.

Reply:
@mkraft This is what I was referring too in the chat. Any comments? Are we okay to proceed? 

I personally guess that the likelihood of having 250,000 groups all enabled for group mention is highly unlikely and if that ever is the case we will need more refactoring that just in this case.

Reply:
Spoke with Martin, we can proceed as is and fix for this if it ever becomes a problem. It doesn't seem like a likely problem as it's unlikely someone will want to enable group mentions on 250K groups or more.